clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/unsecured/OAuthBearerUnsecuredLoginCallbackHandler.java

```diff
@@ -52,6 +57,10 @@
  * name and value except '{@code iat}' and '{@code exp}', both of which are
  * calculated automatically.
  * <p>
+ * <p>
+ * You can also add custom unsecured SASL extensions using
+ * {@code unsecuredLoginExtension_<extensionname>}
```

Comment:
Need to state that the extension name must match th required regex but cannot be the reserved value "auth".